src/gui/application.cpp

```diff
@@ -93,32 +93,25 @@ bool Application::configVersionMigration()
         } else {
             boldMessage = tr("Continuing will mean <b>ignoring these settings</b>.");
         }
-
-        QMessageBox box(
-            QMessageBox::Warning,
-            Theme::instance()->appNameGUI(),
-            tr("Some settings were configured in newer versions of this client and "
-               "use features that are not available in this version.<br>"
-               "<br>"
-               "%1<br>"
-               "<br>"
-               "The current configuration file was already backed up to <i>%2</i>.")
-                .arg(boldMessage, backupFile));
-        box.addButton(tr("Quit"), QMessageBox::AcceptRole);
-        auto continueBtn = box.addButton(tr("Continue"), QMessageBox::DestructiveRole);
-
-        box.exec();
-        if (box.clickedButton() != continueBtn) {
-            QTimer::singleShot(0, qApp, &QApplication::quit);
+        QString message = tr("Some settings were configured in newer versions of this client and "
+                             "use features that are not available in this version.<br>"
+                             "<br>"
+                             "%1<br>"
+                             "<br>"
+                             "The current configuration file was already backed up to <i>%2</i>.")
+                              .arg(boldMessage, backupFile);
+
+        if (!askContinueToDowngrade(message)) {
             return false;
         }
 
         auto settings = ConfigFile::settingsWithGroup(QStringLiteral("foo"));
         settings->endGroup();
 
         // Wipe confusing keys from the future, ignore the others
-        for (const auto &badKey : qAsConst(deleteKeys))
+        for (const auto &badKey : qAsConst(deleteKeys)) {
             settings->remove(badKey);
```

Comment:
I thought we no longer remove random settings but prevent loading the old config?

Reply:
removed it.